### PR TITLE
Browser: Support CSS custom properties

### PR DESCRIPTION
**Pull request by TobyAsE**
This PR adds support for custom CSS properties. I didn't really know about them before, so here's the [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties) and [Spec](https://www.w3.org/TR/css-variables-1/#defining-variables).

There is no error handling at all and the code for the resolution is pretty ugly, but it works.

GitHub has colors (and a lot of greys) when opened in the Browser! :^)

### What is still missing?
- specified fallbacks
- invalid variable fallbacks
- nested `var()`

But I think this is a good starting point.

If anybody wants to test this, [here](https://github.com/SerenityOS/serenity/files/6535104/var.zip) is the file I created to test the implementation (as well as GitHub of course)

----

_GitHub:_
image
_Test Page:_
image

### CODE REVIEW

File: Userland/Libraries/LibWeb/CSS/StyleResolver.cpp [hunk 1]
```diff
@@ -765,6 +765,37 @@ static void set_property_expanding_shorthands(StyleProperties& style, CSS::Prope
     style.set_property(property_id, value);
 }
 
+StyleResolver::CustomPropertyResolutionTuple StyleResolver::resolve_custom_property_with_specificity(const DOM::Element& element, String custom_property_name) const
```
Comment (linusg) [inline Userland/Libraries/LibWeb/CSS/StyleResolver.cpp:768]: Can `custom_property_name` be a const reference? I don't see why we'd have to pass by value, but I might be missing something.

File: Userland/Libraries/LibWeb/CSS/StyleResolver.cpp [hunk 2]
```diff
@@ -765,6 +765,37 @@ static void set_property_expanding_shorthands(StyleProperties& style, CSS::Prope
...
+        auto custom_property_style = match.rule->declaration().custom_property(custom_property_name);
+        if (custom_property_style.has_value()) {
+            return { custom_property_style.value(), match.specificity };
+        }
+    }
+
+    return parent_resolved;
+}
+
+Optional<StyleProperty> StyleResolver::resolve_custom_property(const DOM::Element& element, String custom_property_name) const
```
Comment (linusg) [inline Userland/Libraries/LibWeb/CSS/StyleResolver.cpp:792]: Same as above

File: Userland/Libraries/LibWeb/CSS/StyleResolver.cpp (hunk 1 above)
Comment (TobyAsE) [reply Userland/Libraries/LibWeb/CSS/StyleResolver.cpp]: It probably can. It still works with the modification ;)

File: Userland/Libraries/LibWeb/CSS/Parser/DeprecatedCSSParser.cpp [hunk 3]
```diff
@@ -214,7 +214,19 @@ static CSS::Length parse_length(const CSS::ParsingContext& context, const String
 
 static bool takes_integer_value(CSS::PropertyID property_id)
 {
-    return property_id == CSS::PropertyID::ZIndex || property_id == CSS::PropertyID::FontWeight;
+    return property_id == CSS::PropertyID::ZIndex || property_id == CSS::PropertyID::FontWeight || property_id == CSS::PropertyID::Custom;
+}
+
+static StringView parse_custom_property_name(StringView value)
```
Comment (linusg) [inline Userland/Libraries/LibWeb/CSS/Parser/DeprecatedCSSParser.cpp:220]: No need to copy this either.

```suggestion
static StringView parse_custom_property_name(const StringView& value)
```

Comment (linusg) [reply Userland/Libraries/LibWeb/CSS/Parser/DeprecatedCSSParser.cpp:220]: Turns out this is harmless

**Pull request merged**